webapp/src/ts/modules/reports/reports.component.ts

```diff
@@ -389,20 +391,41 @@ export class ReportsComponent implements OnInit, AfterViewInit, OnDestroy {
     this.reportsActions.selectReport(report);
   }
 
-  selectAllReports() {
-    if (this.reportsList?.length === this.selectedReports?.length) {
+  async selectAllReports() {
+    if (this.areAllReportsSelected()) {
       return;
     }
 
-    this.reportsActions.selectAll();
+    try {
+      if (this.isSidebarFilterOpen) {
+        this.toggleFilter();
+      }
 
-    if (this.isSidebarFilterOpen) {
-      this.toggleFilter();
+      this.globalActions.setLoadingContent(true);
+
+      const summaries = await this.searchService.search(
+        'reports',
+        this.filters,
+        { limit: this.LIMIT_SELECT_ALL_REPORTS, hydrateContactNames: true }
+      );
+      const selected = summaries.map(summary => ({
```

Comment:
I wonder if we can just use the `prepareReports` function that already exists, which handles lineage levels correctly?

Reply:
Almost yes, but I'm afraid I'm missing something important, I've been looking at the code and the `summary` is considered different between both codes.

When selecting all reports, it has `summary` object:
![Screen Shot 2022-11-04 at 8 55 38 am](https://user-images.githubusercontent.com/66472237/199868989-3b92928c-4dab-4ee5-b4d2-3ae100792b93.png)

Then this is what `prepareReports` generates
![Screen Shot 2022-11-04 at 8 57 03 am](https://user-images.githubusercontent.com/66472237/199868982-a01be6e6-924d-482d-8dd0-efe589ca98c2.png)

In this last one the `summary` property is the `form.title` or `report.form`, and is used to display that text [here](https://github.com/medic/cht-core/blob/7778-bulk-delete-redesign/webapp/src/ts/modules/reports/reports.component.html#L62). 

However for selected, the summary as an object is used to check validity and display several values in the DOM, for example [here](https://github.com/medic/cht-core/blob/8f03b189708481c9fae6eeb47ad44cb69d691052/webapp/src/ts/modules/reports/reports-content.component.ts#L81).

In case this is relevant, when selecting 1 report, the object looks like this, it doesn't have `summary` but it has `formatted`:
![Screen Shot 2022-11-04 at 8 54 47 am](https://user-images.githubusercontent.com/66472237/199868995-01dc0a89-b9c2-4117-ab8d-77c63e5f8971.png)

Reply:
I'm not 100% clear why when selecting all, the `summary` is the report itself.

We could do this, I hope I'm not missing important:
```
private async prepareReports(reports) {
    const userLineageLevel = await this.currentLevel;

    return reports.map(report => {
      const form = _find(this.forms, { code: report.form });
      report.icon = form && form.icon;
      report.heading = this.getReportHeading(form, report);

// <The new>
      report.expanded = false;
      report.summaryTitle = form ? form.title : report.form;
      report.summary = report; 
// </>

      report.lineage = report.subject && report.subject.lineage || report.lineage;
      report.unread = !report.read;

      // remove the lineage level that belongs to the offline logged-in user
      if (userLineageLevel && report?.lineage?.length) {
        report.lineage = report.lineage.filter(level => level);
        const item = report.lineage[report.lineage.length -1];
        if (item === userLineageLevel) {
          report.lineage.pop();
        }
      }

      return report;
    });
  }
```
What do you think?

Reply:
Oh my. The summary property being so inconsistent makes me really sad. 
I think that, in list view, we call what appears to be the "subtitle" a summary, and that is consistent across list views. Before 3.10, all list views used the same template. I fear that if we use `summaryTitle` here, we diverge from the common template even further. Also, maybe `subTitle` or `subHeading` would be more self explanatory? 

I wonder if we could pass a parameter to prepare reports (list or content) and just reuse the same property, `summary`, except assign it differently based on the parameter?

Reply:
Cool, I'm going to send summary as parameter in prepareReports

Reply:
done! I like the result thanks for the suggestion!